client/my-sites/themes/themes-banner/index.jsx

```diff
@@ -3,34 +3,105 @@
  */
+
 import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+
+import { localize } from 'i18n-calypso';
 import PropTypes from 'prop-types';
-import cssSafeUrl from 'lib/css-safe-url';
 import Button from 'components/button';
+import safeImageUrl from 'lib/safe-image-url';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { recordTracksEvent } from 'state/analytics/actions';
+import { getActiveTheme, getThemeDetailsUrl } from 'state/themes/selectors';
 
-export class ThemesBanner extends PureComponent {
+class ThemesBanner extends PureComponent {
 	static propTypes = {
-		title: PropTypes.string.isRequired,
-		description: PropTypes.string.isRequired,
-		action: PropTypes.func,
-		actionLabel: PropTypes.string.isRequired,
-		backgroundImage: PropTypes.string,
-		href: PropTypes.string,
+		themeId: PropTypes.string,
+		themeName: PropTypes.string,
+		title: PropTypes.node.isRequired,
+		description: PropTypes.node.isRequired,
+		buttonLabel: PropTypes.string.isRequired,
+		backgroundColor: PropTypes.string,
+		image: PropTypes.string,
+		imageAttrs: PropTypes.shape( {
+			width: PropTypes.number,
+		} ),
+		imageTransform: PropTypes.string,
+
+		// Connected props
+		siteId: PropTypes.number,
+		themeUrl: PropTypes.string,
+		activeThemeId: PropTypes.string,
+	};
+
+	onClick = () => {
+		const { siteId, themeId, activeThemeId } = this.props;
+		const tracksData = {
+			site_id: siteId,
+			theme: themeId,
+			active_theme: activeThemeId,
+		};
+		recordTracksEvent( 'calypso_showcase_banner_click', tracksData );
 	};
 
 	render() {
-		const { title, description, actionLabel, action, backgroundImage, href } = this.props;
-		const backgroundStyle = backgroundImage
-			? { backgroundImage: `url( ${ cssSafeUrl( backgroundImage ) } )` }
-			: null;
+		const {
+			title,
+			description,
+			buttonLabel,
+			backgroundColor,
+			image,
+			imageAttrs = {},
+			imageTransform = 'auto',
+			themeName,
+			themeUrl,
+			translate,
+		} = this.props;
+		const backgroundStyle = backgroundColor ? { backgroundColor } : {};
+		const imageNode = image ? (
+			<img
+				{ ...imageAttrs }
+				alt={ translate( '%(themeName)s Theme', {
+					args: { themeName },
+				} ) }
+				className="themes-banner__image"
+				src={ safeImageUrl( image ) }
+				style={ { transform: imageTransform } }
+			/>
+		) : null;
 		return (
-			<div className="themes-banner" style={ backgroundStyle }>
-				<h1>{ title }</h1>
-				<p>{ description }</p>
-				<Button compact primary onClick={ action } href={ href }>
-					{ actionLabel }
+			<a
+				className="themes-banner"
+				role="button"
+				style={ backgroundStyle }
+				onClick={ this.onClick }
+				href={ themeUrl }
+			>
+				<h1 className="themes-banner__title">{ title }</h1>
+				<p className="themes-banner__description">{ description }</p>
+				<Button className="themes-banner__cta" compact primary>
+					{ buttonLabel }
```

Comment:
As discussed during our call, if we confine the banner to themes (as is reflected in the name we're using now -- `ThemesBanner`) and aren't going to use it for different purposes (such as upgrades/plans etc.), it makes sense to drop the `buttonLabel` prop and just hard-wire it as `translate( 'See the theme' )`.

Reply:
Yeah, that can be done.

What if I make it optional and have it default to `See the theme`?

Reply:
> But, enforcing YAGNI I think it's best to hardcode it for now, if we need a new label then we change accordingly.
> 
> No need to have unnecessary complexity or flexibility, things can change later and evolve as the needs arise.

Yeah, that'd be my line of thinking here. If the need arises later, we can always add the prop (maybe with a default, as you're suggesting). Better to start simple and add complexity only when needed IMO 😄